pkg/cmd/cli/backuplocation/delete.go

```diff
@@ -120,7 +121,65 @@ func Run(f client.Factory, o *cli.DeleteOptions) error {
 			continue
 		}
 		fmt.Printf("Backup storage location %q deleted successfully.\n", location.Name)
+
+		// Delete backups associated with the deleted BSL.
+		backupList, err := findAssociatedBackups(kbClient, location.Name, f.Namespace())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("find backups associated with BSL %q: %w", location.Name, err))
+		}
+		if deleteErrs := deleteBackups(kbClient, backupList); deleteErrs != nil {
+			errs = append(errs, deleteErrs...)
+		}
+
+		// Delete Restic repositories associated with the deleted BSL.
+		resticRepoList, err := findAssociatedResticRepos(kbClient, location.Name, f.Namespace())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("find Restic repositories associated with BSL %q: %w", location.Name, err))
+		}
+		if deleteErrs := deleteResticRepos(kbClient, resticRepoList); deleteErrs != nil {
+			errs = append(errs, deleteErrs...)
+		}
 	}
 
 	return kubeerrs.NewAggregate(errs)
 }
+
+func findAssociatedBackups(client kbclient.Client, bslName, ns string) (velerov1api.BackupList, error) {
+	var backups velerov1api.BackupList
+	err := client.List(context.Background(), &backups, &kbclient.ListOptions{
+		Namespace: ns,
+		Raw:       &metav1.ListOptions{LabelSelector: bslLabelKey + "=" + bslName},
+	})
+	return backups, err
+}
+
+func findAssociatedResticRepos(client kbclient.Client, bslName, ns string) (velerov1api.ResticRepositoryList, error) {
+	var repos velerov1api.ResticRepositoryList
+	err := client.List(context.Background(), &repos, &kbclient.ListOptions{
+		Namespace: ns,
+		Raw:       &metav1.ListOptions{LabelSelector: bslLabelKey + "=" + bslName},
+	})
+	return repos, err
+}
+
+func deleteBackups(client kbclient.Client, backups velerov1api.BackupList) []error {
+	var errs []error
+	for _, backup := range backups.Items {
+		if err := client.Delete(context.Background(), &backup, &kbclient.DeleteOptions{}); err != nil {
+			errs = append(errs, errors.WithStack(fmt.Errorf("delete backup %q associated with deleted BSL: %w", backup.Name, err)))
```

Comment:
Need a `continue` here, or the `Backup associated with deleted BSL(s) %q deleted successfully` will be printed even though we get errors

Reply:
Good catch. I'll add a `continue` after the append errors call on line 169.